jsurfer-core/src/main/java/org/jsfr/json/filter/NotEqualityTypePredicate.java

```diff
@@ -0,0 +1,37 @@
+package org.jsfr.json.filter;
+
+import org.jsfr.json.PrimitiveHolder;
+import org.jsfr.json.path.JsonPath;
+import org.jsfr.json.provider.JsonProvider;
+
+public class NotEqualityTypePredicate extends BasicJsonPathFilter {
```

Comment:
We can unify this class with `EqualityTypePredicate` with some `boolean invert` field

Reply:
A few reasons why better to not do it:
1. Performance: we create different instances of different classes once on compilation time whereas this `if-check` will consume CPU for every JSON element in the input data.
2. Breaking the single responsibility principle.
3.  Bringing different style that differs from currently existing other predicates.